test/acceptance/user.controller.acceptance.ts

```diff
@@ -0,0 +1,132 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/example-shopping
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {createClientForHandler, supertest, expect} from '@loopback/testlab';
+import {RestServer} from '@loopback/rest';
+import {ShoppingApplication} from '../..';
+import {UserRepository} from '../../src/repositories';
+import {UserDataSource} from '../../src/datasources';
+
+describe('UserController', () => {
+  let app: ShoppingApplication;
+  let server: RestServer;
+  let client: supertest.SuperTest<supertest.Test>;
+  const userRepo = new UserRepository(new UserDataSource());
+
+  const user = {
```

Comment:
I'd prefer us to follow our own best practices, see [Use data builders](https://loopback.io/doc/en/lb4/Testing-your-application.html#use-test-data-builders).